hoomd/hpmc/update.py

```diff
@@ -15,387 +15,141 @@
-        Rescale aspect ratio along a randomly chosen dimension.
+        * ``volume``: `tuple` [`int`, `int`] - Number of accepted and rejected volume and length moves.
+        * ``shear``: `tuple` [`int`, `int`] - Number of accepted and rejected shear moves.
+        * ``aspect``: `tuple` [`int`, `int`] - Number of accepted and rejected aspect moves.
 
         Note:
-            When an argument is None, the value is left unchanged from its current state.
-
-        Example::
-
-            box_update.aspect(delta=0.01)
-            box_update.aspect(delta=0.01, weight=2)
-            box_update.aspect(delta=0.01, weight=0.15)
-
-        Returns:
-            A :py:class:`dict` with the current values of *delta*, and *weight*.
-
-        """
-        self.check_initialization();
-
-        if weight is not None:
-            self.aspect_weight = float(weight)
-
-        if delta is not None:
-            self.aspect_delta = float(delta)
-
-        self.cpp_updater.aspect(self.aspect_delta, self.aspect_weight);
-        return {'delta': self.aspect_delta, 'weight': self.aspect_weight}
-
-    def get_volume_acceptance(self):
-        R""" Get the average acceptance ratio for volume changing moves.
-
-        Returns:
-            The average volume change acceptance for the last run
-
-        Example::
-
-            mc = hpmc.integrate.shape(..);
-            mc.shape_param[name].set(....);
-            box_update = hpmc.update.boxmc(mc, betaP=10, seed=1)
-            run(100)
-            v_accept = box_update.get_volume_acceptance()
-
+            The counts are reset to 0 at the start of each
+            `hoomd.Simulation.run`.
         """
-        counters = self.cpp_updater.getCounters(1);
-        return counters.getVolumeAcceptance();
-
-    def get_ln_volume_acceptance(self):
-        R""" Get the average acceptance ratio for log(V) changing moves.
-
-        Returns:
-            The average volume change acceptance for the last run
-
-        Example::
-
-            mc = hpmc.integrate.shape(..);
-            mc.shape_param[name].set(....);
-            box_update = hpmc.update.boxmc(mc, betaP=10, seed=1)
-            run(100)
-            v_accept = box_update.get_ln_volume_acceptance()
-
-        """
-        counters = self.cpp_updater.getCounters(1);
-        return counters.getLogVolumeAcceptance();
-
-    def get_shear_acceptance(self):
-        R"""  Get the average acceptance ratio for shear changing moves.
-
-        Returns:
-           The average shear change acceptance for the last run
-
-        Example::
-
-            mc = hpmc.integrate.shape(..);
-            mc.shape_param[name].set(....);
-            box_update = hpmc.update.boxmc(mc, betaP=10, seed=1)
-            run(100)
-            s_accept = box_update.get_shear_acceptance()
+        if not self.is_attached:
+            return None
+        else:
+            return self._cpp_obj.getCounters(1)
 
+    @log(flag="sequence")
+    def volume_moves(self):
```

Comment:
The `log` decorator makes methods properties by default. This is to prevent in nearly all cases,
```
@property 
@log 
def foo(self):
    pass
```

Reply:
In that case, should this be `log_property` or something like that? Most of the property-like decorators I know have property in the name, such as `abstractproperty`. The word “log” alone doesn’t convey the other crucial meaning here.

Reply:
We could create a separate method `log_property`. Right now `log` accepts a key-word only argument `is_property` which defaults to `True`.